lib/compiler/test/bs_match_SUITE.erl

```diff
@@ -1749,6 +1751,18 @@ do_erl_689_2b(_, <<Length, Data/binary>>) ->
             {{Y, M, D}, Rest}
     end.
 
+%% ERL-753
+
+bs_start_match2_defs(_Config) ->
+    _ = api_url({https, "hello"}, dummy).
```

Comment:
I would suggest that you test the values returned from the function (calling the function twiced, once with a binary and once with a tuple).

The reason is that a bug fix might fix the immediate problem (crashing or failing to pass `beam_validator`), but the compiler might still produce incorrect code (now or in the future).

Reply:
Thanks, I've force-pushed a fix.